snuba/migrations/snuba_migrations/sessions/0002_sessions_aggregates.py

```diff
@@ -0,0 +1,178 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import (
+    AggregateFunction,
+    Column,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+DEFAULT_QUANTITY = 1
+
+
+class Migration(migration.MultiStepMigration):
+    """
+    This is the first migration that adds new columns to the raw and materialized
+    sessions dataset.
+    A second migration will be added later that updates the materialization query
+    to use the `quantity` raw column and feed that into the `X_sum` columns.
+    """
+
+    blocking = False
+
+    def forwards_local(self) -> Sequence[operations.Operation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.SESSIONS,
+                table_name="sessions_raw_local",
+                column=Column(
+                    "quantity", UInt(32, Modifiers(default=str(DEFAULT_QUANTITY)))
+                ),
+                after="distinct_id",
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.SESSIONS,
+                table_name="sessions_hourly_local",
+                column=Column(
+                    "duration_avg", AggregateFunction("avgIf", [UInt(32), UInt(8)])
+                ),
+                after="duration_quantiles",
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.SESSIONS,
+                table_name="sessions_hourly_local",
+                column=Column(
+                    "sessions_sum", AggregateFunction("sumIf", [UInt(32), UInt(8)])
+                ),
+                after="sessions",
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.SESSIONS,
+                table_name="sessions_hourly_local",
+                column=Column(
+                    "sessions_crashed_sum",
+                    AggregateFunction("sumIf", [UInt(32), UInt(8)]),
+                ),
+                after="sessions_crashed",
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.SESSIONS,
+                table_name="sessions_hourly_local",
+                column=Column(
+                    "sessions_abnormal_sum",
+                    AggregateFunction("sumIf", [UInt(32), UInt(8)]),
+                ),
+                after="sessions_abnormal",
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.SESSIONS,
+                table_name="sessions_hourly_local",
+                column=Column(
+                    "sessions_errored_sum",
+                    AggregateFunction("sumIf", [UInt(32), UInt(8)]),
+                ),
+                after="sessions_errored",
+            ),
+        ]
```

Comment:
What is the reason for not having the materialized view being rebuilt here?

Reply:
doing that in https://github.com/getsentry/snuba/pull/1459
I thought we agreed to do it step-by-step, especially since the matview will write to the newly created columns, and we need support to query those first

Reply:
I think there is a misunderstanding here. 

> ... will write to the newly created columns, and we need support to query those first

This is correct, but from the consumer. The materialized view can be changed to start filling new columns. 
At this point, since no aggregated session is being ingested, the materialized view will end up writing nothing to the new columns anyway. And, since the old columns are still intact and being filled properly the destination table is backward compatible. 
Of course the materialized view must be backward compatible and still keep filling in old data in the old columns. But this has to happen no matter if the matview is recreated here or in a following PR since there is no way to prform the matview update and the deployment atomically.

Reply:
> the materialized view must be backward compatible and still keep filling in old data in the old columns

I think I can make that work. Since the new aggregated entries will have a NIL session_id per definition, we can put that into the if condition to fill the `count` column for non-aggregated entries, and the `sum` column for the new aggregated ones.